components/report-manager-service/worker/worker.go

```diff
@@ -248,6 +249,28 @@ func (t *GenerateReportTask) Run(ctx context.Context, task cereal.Task) (interfa
 			return nil, err
 		}
 	}
+	lifecycles, err := t.ObjStoreClient.GetBucketLifecycle(ctx, job.RequestToProcess.RequestorId)
+	// fmt.Println(lifecycles)
+
+	if lifecycles == nil && err.Error() == "The lifecycle configuration does not exist" {
```

Comment:
This code executes only if the bucket doesn't have any life cycle rules. What if the bucket has life cycle rules, and doesn't have any expiration. It is possible that bucket can have object transition rules and in these cases, we have to create an expiration rule as well.